source/manuals/programming-languages/nodejs/style-guide.html.md.erb

```diff
@@ -0,0 +1,119 @@
+---
+title: GDS Node.js Style Guide
+---
+
+# <%= current_page.data.title %>
+
+## General style
+
+As with all JavaScript at GDS, for Node.js applications we use the [standard] style.
+
+## Use strict mode
+
+[Strict mode] should be enabled for Node.js modules, as in the browser this prevents polluting the global scope (in
+Node.js this would pollute the `global` object, rather than the `window` object).
+
+## ES2015+
+
+Because with Node.js you control the version of the interpreter you're code is running on,
+you have access to many the latest features of JavaScript, such as:
+
+- [let] and [const]
+- [Arrow functions]
+- [Array.prototype] methods like `Array.map()`, `Array.find()` and `Array.values()`
+- The [spread operator]
+
+However use caution when using cutting edge JavaScript features and be sure to read the [documentation].
+
+A full list of what features are supported in each version of Node.js can be found at [Node Green].
+
+
+### let and const
+
+For Node.js code, be explicit about variable mutability using [const] and [let]:
+
+```
+const foo // GOOD: explicitly immutable variable declaration
+let bar // GOOD: expicitly mutable variable declaration
+var baz // BAD: not explicit about mutability of variable
+```
+
+This makes code easier to understand, as well as providing a feedback loop if a [const] variable is inadvertently reassigned.
+
+### Promises
+
+For Node.js applications, we use the native `Promise` implementation available in Node.js over alternatives like `q` or `bluebird`.
```

Comment:
Is it worth adding details about why an alternative was needed and what version of Node.js started to support native Promises? Just for some context around this point.

Reply:
Added bit on when they've been supported from. I didn't want to go too much into the history of it in the style guide. Though I ought to do a blog post on it.